pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -77,6 +77,11 @@ func (configgen *ConfigGeneratorImpl) buildGatewayListeners(env model.Environmen
 		// When Envoy supports filter chain matching, we'll have to group the ports by number and protocol, so this logic will
 		// no longer work.
 		protocol := model.ParseProtocol(servers[0].Port.Protocol)
+		if protocol == model.ProtocolHTTPS {
+			// Gateway terminates TLS connection. So, its effectively a H2 listener.
```

Comment:
Huh? Why does this work? I thought the protocols were binary incompatible (i.e. we can't process HTTP with the H2 filter, right?). Either way I don't really follow the comment.

Reply:
Http connection manager autodetects h1 or h2. The gateway https is same as ingress https port. Terminate tls if mode is not passthru. I will update code n comment.